### DISCUSSION TITLE
Examples type safety not great in Typescript

### DISCUSSION THREAD

**Discussion by mikecann** [reimagined/resolve#1968] [General]
Hey guys, really appreciate you putting the time into making some typescript examples, sorry it has taken me so long to exploring them.

Although there are now some types it seems like there is an awful lot of "any" spread around the codebase. 

For [example](https://github.com/reimagined/resolve/blob/dev/examples/ts/hacker-news/common/aggregates/user.commands.ts#L9) none of the payloads are typed so everything is "any". Same goes for the aggregate state. 

If we set `"noImplicitAny": true` and `"strictNullChecks": true` (which should be the minimum for any modern TS project, ideally you should be using `"strict": true`) in the tsconfig.json we get all kind of errors. 

Currently as it stands there is little benefit to the TS examples over the JS ones.

**Comment by const314** [comment]
We are working on making reSolve fully typed. We still have some packages to convert to typescirpt first, so typing demos will take some time.
As for the case you mentioned, using 'any' for payloads type may be reasonable, at least if they are not used anywhere outside of the aggregate's commands and projections.
Let's convert this issue to a discussion and discuss the scenarios where types are mandatory in demos.

**Reply by mikecann** [reply]
> We are working on making reSolve fully typed. 

I appreciate that and applaud you for your efforts. I was meerly checking out the library as a response to https://github.com/reimagined/resolve/discussions/1750#discussioncomment-977638 and was a little disappointed to see that it wasnt really ready for prime time yet.

> As for the case you mentioned, using 'any' for payloads type may be reasonable

Im not sure I understand why this would be reasonable. Type safety in the payloads is one of the places I would want it most TBH. Any sort of implict any or lack of type-saftey (outside of very tightly confined scopes) leads to issues in development and refactoring im my experience.

**Reply by const314** [reply]
Easier refactoring was our main reason to convert the entire framework to typescript.
As for the demos, I think we got your point.
Typing everything in demos seems reasonable if we want to show the best approach to make a reSolve application in typescript.

**Reply by mikecann** [reply]
@const314 Thankyou for your swift reply. Apologies if I sound like an ungrateful so-and-so. I realise this is open-source software and you guys have lots of different priorities I am very excited about the potential of this project. Please do let me know if there is anything I can do to help.
